src/BugsnagUnity/Native/Android/NativeInterface.cs

```diff
@@ -1200,5 +1211,22 @@ public void ClearFeatureFlags()
             AndroidJNI.CallVoidMethod(GetClientRef(), ClearFeatureFlagsMethod, null);
         }
 
+        public void RegisterForOnSessionCallbacks()
+        {
+            // This was added to android only because we suspect session callbacks might be causing ANRs.
+            // Not many users use session callbacks, so we are only subscribing to the native side if necessary 
```

Comment:
Can you add this to the `INativeClient` definition – although I think the lasting justification should be that it's there as an optimisation to avoid unnecessary calls to the native layer if not required. The ANR driver should result in a separate fix.